examples/log.c

```diff
@@ -337,17 +338,57 @@ static void print_time(const git_time *intime, const char *prefix)
 }
 
 /** Helper to print a commit object. */
-static void print_commit(git_commit *commit)
+static void print_commit(git_commit *commit, struct log_options options)
 {
 	char buf[GIT_OID_HEXSZ + 1];
 	int i, count;
 	const git_signature *sig;
 	const char *scan, *eol;
+	unsigned long length = 0;
 
 	git_oid_tostr(buf, sizeof(buf), git_commit_id(commit));
 	printf("commit %s\n", buf);
 
-	if ((count = (int)git_commit_parentcount(commit)) > 1) {
+	sig = git_commit_author(commit);
+	scan = git_commit_message(commit);
+	count = (int)git_commit_parentcount(commit);
+
+	if (options.show_log_size) {
+
+		if (sig != NULL) {
+
+			length += strlen(sig->name);
+			length += strlen(sig->email);
+
+			/* The date field (and other fixed-width fields) are added as
+			 * constants 
+			 */
+			length += 13;
+			length += 39;
+
+		}
+
+		if (count > 1)
+			length += 8*count + 12;
+
+
+		/* simulate the message-identation */
+		while (scan && *scan) {
+
+			if (*scan == '\n')
+				length += 4;
+			scan++;
+
+		}
```

Comment:
All of the above looks quite a lot like guessworking. Wouldn't it be a lot easier if you just addded the return value of all those `printf`s to `length`? That'd match exactly all the bytes we're outputting, and make the logic much easier to follow.